src/pudl/transform/ferc1.py

```diff
@@ -7,1400 +7,763 @@
+
+        This is necessary because the fuel type is a messy freeform string column that
+        needs to be cleaned up, and is also (insanely) a primary key column for the
+        table, and required for merging the fuel_ferc1 and plants_steam_ferc1 tables.
+        This means that we can't assign a record ID until the fuel types have been
+        cleaned up. Additionally the string categorization results in a number of
+        duplicate fuel records which need to be aggregated.
+        """
+        return (
+            self.merge_instant_and_duration_tables_xbrl(
+                raw_xbrl_instant, raw_xbrl_duration
+            )
+            .pipe(self.rename_columns, params=self.params.rename_columns_ferc1.xbrl)
+            .pipe(self.convert_units_multicol, params=self.params.convert_units)
+            .pipe(self.normalize_strings_multicol, params=self.params.normalize_strings)
+            .pipe(
+                self.categorize_strings_multicol, params=self.params.categorize_strings
+            )
+            .pipe(self.standardize_physical_fuel_units)
+            .pipe(self.aggregate_duplicate_fuel_types_xbrl)
+            .pipe(self.assign_utility_id_ferc1_xbrl)
+            .pipe(self.assign_record_id, source_ferc1=Ferc1Source.XBRL)
+        )
+
+    def standardize_physical_fuel_units(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Convert reported fuel quantities to standard units depending on fuel type.
+
+        Use the categorized fuel type and reported fuel units to convert all fuel
+        quantities to the following standard units, depending on whether the fuel is a
+        solid, liquid, or gas. When a single fuel reports its quantity in fundamentally
+        different units, convert based on typical values. E.g. 19.85 MMBTU per ton of
+        coal, 1.037 Mcf per MMBTU of natural gas, 7.46 barrels per ton of oil.
+
+          * solid fuels (coal and waste): short tons [ton]
+          * liquid fuels (oil): barrels [bbl]
+          * gaseous fuels (gas): thousands of cubic feet [mcf]
+
+        Columns to which these physical units apply:
+
+          * fuel_consumed_units (tons, bbl, mcf)
+          * fuel_cost_per_unit_burned (usd/ton, usd/bbl, usd/mcf)
+          * fuel_cost_per_unit_delivered (usd/ton, usd/bbl, usd/mcf)
+
+        One remaining challenge in this standardization is that nuclear fuel is reported
+        in both mass of Uranium and fuel heat content, and it's unclear if there's any
+        reasonable typical conversion between these units, since available heat content
+        depends on the degree of U235 enrichement, the type of reactor, and whether the
+        fuel is just Uranium, or a mix of Uranium and Plutonium from decommissioned
+        nuclear weapons. See:
+
+        https://world-nuclear.org/information-library/facts-and-figures/heat-values-of-various-fuels.aspx
+
+        """
+        df = df.copy()
+
+        FuelFix = namedtuple("FuelFix", "fuel from_unit to_unit mult")
```

Comment:
is `FuelFix = namedtuple("FuelFix", ["fuel", "from_unit", "to_unit", "mult"])` the same? I find the string w/ spaces turning into multiple items here a lil hard to read.

Reply:
Ah, [so it is](https://docs.python.org/3/library/collections.html#collections.namedtuple)! I guess I was just using a weird example as a template.

Reply:
Fixed.